File: sources/Variance2.py
from typing import Sequence


class Fruit:
    pass


class Orange(Fruit):
    def __repr__(self):
        return "Orange"


class Apple(Fruit):
    def __repr__(self):
        return "Apple"


def printContent(punnet : Sequence[Fruit]):
    for Fruit in punnet:
        print(Fruit)


punnet : Sequence[Orange] = [Orange(), Orange(), Orange()]

printContent(punnet)

File: sources/Variance2.java
class Fruit {
}

class Orange extends Fruit {
    public String toString() {
        return "Orange";
    }
}

class Apple extends Fruit {
    public String toString() {
        return "Apple";
    }
}

public class Variance2 {
    public static void smichej(Fruit[] punnet) {
        punnet[0] = new Orange();
        punnet[1] = new Apple();
    }

    public static void main(String[] args) {
        Fruit[] punnet = new Orange[2];
        smichej(punnet);

        for (Fruit Fruit:punnet) {
            System.out.println(Fruit);
        }
    }
}
